password/impl/src/main/java/org/wildfly/security/password/impl/OneTimePasswordImpl.java

```diff
@@ -137,11 +162,17 @@ private static byte[] generateOTP(String algorithm, byte[] passPhrase, String se
         byte[] hash = hashAndFold(algorithm, messageDigest, seedAndPassPhrase.toArray());
 
         // Computation step
-        for (int i = 0; i < sequenceNumber; i++) {
+        hash = updateSequence(hash, algorithm, messageDigest, 0, sequenceNumber);
+        return hash;
+    }
+
+    static byte[] updateSequence(final byte[] hash, final String algorithm, final MessageDigest messageDigest, final int currentSequenceNumber, final int newSequenceNumber) {
```

Comment:
Maybe this could be renamed to something like `computeHash`?

Reply:
And perhaps the params could be something like `start` and `end` instead of `currentSequenceNumber` and `newSequenceNumber` (just to make things a bit more clear in the `verify` method.